midp/gfx.js

```diff
@@ -357,20 +479,19 @@ var currentlyFocusedTextEditor;
         this.baseline = size | 0;
         this.height = (size * 1.3) | 0;
 
-        this.context = document.createElement("canvas").getContext("2d");
-        this.context.canvas.width = 0;
-        this.context.canvas.height = 0;
-        this.context.font = style + size + "px " + face;
         this.size = size;
+        // XXX Figure out why this works when these two fields are both int.
         this.style = style;
         this.face = face;
     };
 
-    function calcStringWidth(font, str) {
+    function calcStringWidth(font, str, scaleByDPR) {
         var emojiLen = 0;
+        var fontSize = font.size * (scaleByDPR ? DPR : 1);
 
-        var len = font.context.measureText(str.replace(emoji.regEx, function() {
-            emojiLen += font.size;
+        tempContext.font = font.style + " " + fontSize + "px " + font.face;
+        var len = tempContext.measureText(str.replace(emoji.regEx, function() {
+            emojiLen += fontSize;
```

Comment:
I think `calcStringWidth` is called a lot; creating the font every time hasn't created a performance penalty?